src/main/java/org/miracum/streams/ume/obdstofhir/mapper/ObdsProcedureMapper.java

```diff
@@ -500,31 +501,36 @@ public Procedure createRadiotherapyProcedure(
     return stProcedure;
   }
 
+  /**
+   * This will return the whole timespan of completed partial radiations as part of a therapy. If no
+   * usable start or end dates have been found, this method will return a tuple containing `null`
+   * values.
+   *
+   * @param partialRadiations The partial radiations
+   * @return A tuple with start date and end date of all given partial radiations
+   */
   public Tupel<Date, Date> getTimeSpanFromPartialRadiations(List<Bestrahlung> partialRadiations) {
-    // min Beginndatum
-    List<Date> minDates = new ArrayList<>();
-    // maxBeginndatum
-    List<Date> maxDates = new ArrayList<>();
-
-    for (var radio : partialRadiations) {
-      if (radio.getST_Beginn_Datum() != null) {
-        minDates.add(convertObdsDateToDateTimeType(radio.getST_Beginn_Datum()).getValue());
-      }
-      if (radio.getST_Ende_Datum() != null) {
-        maxDates.add(convertObdsDateToDateTimeType(radio.getST_Ende_Datum()).getValue());
-      }
-    }
-
-    Date minDate = null;
-    Date maxDate = null;
-
-    if (!minDates.isEmpty()) {
-      minDate = Collections.min(minDates);
-    }
-    if (!maxDates.isEmpty()) {
-      maxDate = Collections.min(maxDates);
+    if (null == partialRadiations || partialRadiations.isEmpty()) {
+      return new Tupel<>(null, null);
     }
 
-    return new Tupel<>(minDate, maxDate);
+    val minDates =
```

Comment:
```suggestion
    var minDates =
```
ginge auch `var` ? wenn möglich würde ich die dependency auf lombok's `val` vermeiden wollen.